website/pages/en/cookbook/transfer-to-the-graph.mdx

```diff
@@ -0,0 +1,167 @@
+---
+title: Tranfer to The Graph
+---
+
+Learn how to quickly upgrade your subgraphs from any platform to [The Graph's decentralized network](https://thegraph.com/networks/).
+
+## Benefits of Switching to The Graph
+
+- Use the same subgraph that your apps already use with zero-downtime migration.
+- Increase reliability from a global network supported by 100+ Indexers.
+- Receive lightning-fast support for subgraphs 24/7, with an on-call engineering team.
+
+## Upgrade Your Subgraph to The Graph in 3 Easy Steps
+
+1.  [Set Up Your Studio Environment](/cookbook/transfer-to-the-graph/#1-set-up-your-studio-environment)
+2.  [Deploy Your Subgraph to Studio](/cookbook/transfer-to-the-graph/#2-deploy-your-subgraph-to-studio)
+3.  [Publish to The Graph Network](/cookbook/transfer-to-the-graph/#publish-your-subgraph-to-the-graphs-decentralized-network)
+
+## 1. Set Up Your Studio Environment
+
+### Install the Graph CLI⁠
+
+On your local machine, run the following commands:
+
+Using [npm](https://www.npmjs.com/):
+
+```sh
+graph init --product subgraph-studio
+```
+
+> Alternatively, you can set up your environment using codespaces on your Github repo.
+
+### Create a Subgraph in Subgraph Studio
+
+- Go to [Subgraph Studio](https://thegraph.com/studio/) and connect your wallet.
+- Click "Create a Subgraph". It is recommended to name the subgraph in Title Case: "Subgraph Name Chain Name".
+
+> Note: After publishing, the subgraph name will be editable but requires signing each time, so name it properly.
+
+For additional information on subgraph creation and the Graph CLI, check out the subgraph [Quick Start](/quick-start/) and [Creating a Subgraph](/developing/creating-a-subgraph).
+
+### Authenticate Your Subgraph
+
+In The Graph CLI, use the auth command seen in Subgraph Studio:
+
+```sh
+graph auth --studio <your-auth-code>
+```
+
+## 2. Deploy Your Subgraph to Studio
+
+In The Graph CLI, run the following command:
+
+```sh
+graph deploy --studio <slug> --ipfs-hash <your-subgraph-ipfs-hash>
```

Comment:
How do they know the IPFS hash of their subgraph?

Here I think we need two paths:
1. They have their subgraph source code already locally and can build it (e.g. `graph compile`). I assume most devs have this.
2. They don't have access to the source code. Then the `--ipfs-hash` hack makes sense but we need to explain how to get that one. The solution is to find the GraphiQL Playground and send this query:

```
{ _meta { deployment } }
```

The problem is, that it is not guaranteed that this command works, depending on the setup of the centralized competitor. I did not check all edge-cases, so maybe have a info that explains this.

Reply:
Goldsky & Alchemy both have the IPFS hash readily copy-able in each subgraph. They just refer to it as deployment ID. 

Perhaps we just add a little note on how to get the IPFS hash right after showing the command?
> Every subgraph has an IPFS hash, a.k.a. Deployment ID. It starts with "Qm" and looks like"Qmasdfadsfaf". 

I think no need to go too deep into the source code route because those devs already know how to deploy. How about just a brief mention at the start saying:

> If you have your source code, you can easily build & deploy it to Studio. But if not, here's a quick way...